go/cmd/orchestrator-agent/main.go

```diff
@@ -33,11 +33,22 @@ var AppVersion string
 
 func acceptSignal() {
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill, syscall.SIGHUP)
 
 	// Block until a signal is received.
-	sig := <-c
-	log.Fatalf("Got signal: %+v", sig)
+	signal.Notify(c, syscall.SIGHUP, syscall.SIGKILL, syscall.SIGTERM, syscall.SIGTERM)
+	go func() {
+		for sig := range c {
+			switch sig {
+			case syscall.SIGHUP:
+				log.Infof("Received SIGHUP. Reloading configuration")
+				config.Reload()
+			case syscall.SIGTERM, syscall.SIGKILL, syscall.SIGINT:
+				log.Infof("Received %s. Shutting down orchestrator-agent", sig.String())
+				// probably should poke other go routines to stop cleanly here ...
```

Comment:
If you poke goroutines there's nothing to promise they will complete before exiting.